Watchman.Discord/Areas/Announcements/Controllers/AnnouncementsController.cs

```diff
@@ -0,0 +1,41 @@
+﻿using System;
+using System.Threading.Tasks;
+using Discord.WebSocket;
+using Watchman.Discord.Framework.Architecture.Controllers;
+using Watchman.Integrations.Disboard;
+
+namespace Watchman.Discord.Areas.Announcements.Controllers
+{
+    class AnnouncementsController : ReadAlways, IController
+    {
+        private TimeSpan HowOften => new TimeSpan(0, 0, 15); // every 2 hours and 1 minute
+        private bool _autoBumpIsOn;
+        private ServerBumper _bumper;
+
+        [DiscordCommand("-autobump start")]
+        public void AutoBumpStart(SocketMessage socketMessage)
+        {
+            _autoBumpIsOn = true;
+            socketMessage.Channel.SendMessageAsync("Started auto bumping.");
+
+            _bumper = new ServerBumper(socketMessage.Channel);
```

Comment:
nie rób tego przez konstruktor, tylko ustawiaj przez metode

Reply:
ale że co? tu nie ma konstruktora

Reply:
new ServerBumper(socketMessage.Channel);

przekazujesz to przez konstruktor

Reply:
dobrze gdyby bump serwer trzymał liste kanałów 
bo kanały mogą być na róznych serwerach

Reply:
A w czym gorszy jest konstruktor?

Reply:
że będzie tylko jedna instancja obiektu w całym istnieniu bota 
to znaczy
bot będzie mógł obsługiwać tylko jeden kanał w ciągu całego swojego istnienia 

a my chcemy żeby dużo serwerów mogło być obsługiwanych jednocześnie

Reply:
Zamiarem moim było to, żeby dla każdego serwera był osobny obiekt. Dokładnie to dla każdej akcji autobump osobny obiekt ServerBumper, który miałby przypisany kanał. W momencie użycia _autobump stop_ jest porzucany.

Reply:
no tak
ale w momencie tworzenia nowego obiektu, na innym serwerze, stary obiekt jest usuwany 
kontrolery są tworzone raz dla całej aplikacji 
a nie per serwer / per kanał 

dlatego lepiej będzie, jeśli będziesz trzymać liste kanałów, na które chcesz wysyłać powiadomienia i operować na tej liście

Reply:
jak do obecnej referencji przypiszesz nową instancje
to stara jest usuwana

teraz tworzysz nową instancje za każdym razem jak ktoś chce dodać kanał do listy

Reply:
powinna być jedna instancja która trzyma wiele kanałów